articles/unofficial-faq.html

```diff
@@ -1051,24 +1051,25 @@ <h3>
 </h3>
 
 <p>
-In Go, a map is designed as a container which can contain unlimited number of entries if memory is available.
-And, in the official Go runtime implementation, to ensure good map element indexing efficiency,
-each map value only maintains one continuous memory segment for the entire entries stored in that map.
-Therefor, Go runtime needs to allocate larger memory segments for a map from time to time
-when there are more and more entries being put into the map.
+The first reason is that it would conflict with maps' internal memory management process.
+In Go, a map is designed as a container which can contain an unlimited number of entries if memory is available.
+To ensure good map element indexing efficiency, in the official Go runtime implementation 
+each map value only maintains one continuous memory segment for the entirety of entries stored in that map.
+Therefore, Go runtime needs to allocate larger memory segments for a map from time to time
+when more and more entries are being put into the map.
 In the process, the entries stored on older memory segments will be moved to newer memory segments.
-There might be also some other reasons causing entry memory movements.
+There might be also some other reasons causing memory movements of entries.
 In other words, the addresses of map elements will change from time to time on need.
-If map elements are allowed to be taken addresses, then when some map entries are moved,
-Go runtime must update all pointers which are storing the addresses of the moved elements,
-which brings many difficulties in implemnting Go compilers and runtimes and decreases Go program running performance much.
-So, currently, map elements are disallowed to be taken addresses. 
+If map elements are allowed to have their addresses taken, then when some map entries are moved,
+Go runtime would have to update all pointers which are storing the addresses of the moved elements,
+which brings many difficulties in implementing Go compilers and runtimes and greatly decreases execution performance.
+So, currently, map elements cannot have their addresses taken. 
 </p>
 
 <p>
-Another reason why map elements are unaddressable is the map index expression <code>aMap[key]</code> might
+Secondly, the map index expression <code>aMap[key]</code> might
 return an element stored in map <code>aMap</code> or not, which means <code>aMap[key]</code> might still result
-a zero value after <code>(&amp;aMap[key]).Modify()</code> is called. This will confuse many people.
+in a zero value after <code>(&amp;aMap[key]).Modify()</code> is called. This would confuse many people.

```

Comment:
I have no idea what (&aMap[key]).Modify() is supposed to do, just to put that out there. Maybe you explain elsewhere.